node_modules/oae-core/topnavigation/js/topnavigation.js

```diff
@@ -67,7 +67,8 @@ define(['jquery', 'oae.core', 'activityadapter'], function($, oae, ActivityAdapt
         var initLeftMenu = function() {
             var $template = oae.data.me.anon ? $('#topnavigation-left-anonymous-template') : $('#topnavigation-left-loggedin-template');
             oae.api.util.template().render($template, {
-                'authStrategyInfo': authStrategyInfo
+                'authStrategyInfo': authStrategyInfo,
+                'redirectUrl': signInRedirectUrl || oae.api.util.url().attr('relative')
```

Comment:
We have to distinguish between there being an explicit redirect (e.g., a `?url=...` parameter) and us just "softly" indicating where to redirect after login. The former results in the signin action automatically being triggered.